api/src/wfl/module/covid.clj

```diff
@@ -58,16 +31,24 @@
       (throw (ex-info (str "No such record") {:id id :table table})))
     record))
 
-;; interfaces
-;; queue operations
-(defmulti peek-queue!
-  "Peek the first object from the `queue`, if one exists."
-  (fn [queue] (:type queue)))
-
-(defmulti pop-queue!
-  "Pop the first object from the `queue`. Throws if none exists."
-  (fn [queue] (:type queue)))
+(defn peek-queue!
+  "Get first unconsidered item in DETAILS table."
+  [{:keys [details] :as _queue}]
```

Comment:
I found that the queue operations didn't differ based on whether we were checking a source, executor, or sink queue.

As defined, these operations are specific to our model where a `details` table is the queue to check.  We could switch this back to a more generic `queue` check but would need to supplement source/executor/sink with this key-val at load.  Or we accept instead the table to monitor rather than a destructured map.  I think this works for now but can change if there are strong feelings.

Reply:
Ugh.  Yes, thanks for this sanity check.  Will fix.